account/init_tenant.go

```diff
@@ -95,14 +95,19 @@ func CleanTenant(ctx context.Context, config tenantConfig, remove bool) error {
 	}
 	defer rest.CloseResponse(res)
 
-	if res.StatusCode != http.StatusOK {
+	// operation failed for some reason
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
 		jsonErr, err := c.DecodeJSONAPIErrors(res)
 		if err == nil {
+			// use the status code to produce an error
 			if len(jsonErr.Errors) > 0 {
-				return errors.NewInternalError(ctx, fmt.Errorf(jsonErr.Errors[0].Detail))
+				return errors.FromStatusCode(res.StatusCode, jsonErr.Errors[0].Detail)
+			} else {
+				return errors.FromStatusCode(res.StatusCode, rest.ReadBody(res.Body))
```

Comment:
Not sure this will work? Wouldn't res already be read during c.DecodeJSONAPIErrors(res)?

What if err != nil?